modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/resources/META-INF/mappings/liferay-type-mappings.json

```diff
@@ -1870,6 +1870,16 @@
 			"store": true,
 			"type": "keyword"
 		},
+		"urlTitle": {
+			"fields": {
+				"keyword": {
+					"store": true,
```

Comment:
Hi @ericyanLr,

From looking at portal code, urlTitle is only ever added as a Keyword, for example: https://github.com/liferay/liferay-portal/blob/c0e41ad33c7ab8dd960171743f16e40b8522d7d3/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/search/spi/model/index/contributor/JournalArticleModelDocumentContributor.java#L182-L186

If this is the case, we should simply index the urlTitle as a keyword with no sub-field (and reference directly as the filter field).

Reply:
Hi @joshuacords ,

One big difference between the urlTitle and it's associated sortable field, is that the sortable field always stored it's value in lowercase.

In regards to how Headless API allows filtering with the usage of Wildcard Query, my understanding is that it would be case sensitive. The current behavior of Headless API's filtering, relies on the matching term being lowercase.
- Reference: [LPS-85266](https://issues.liferay.com/browse/LPS-85266)

And so, if we switch to using the urlTitle (even though it's a keyword type), since it's not normalized to lowercase, the filtering would no longer work properly.

If you think that we should proceed with this change, can you provide guidance on how filtering on the Headless API side, can workaround this?

I'll also provide more details of my findings below, in case it's helpful.

Please let me know what you think.
Thanks!

----

**Search**
Although urlTitle is a **keyword** type, the value that is stored is not normalized to be lowercase: 
- When calling the method: [document.addKeywordSortable](https://github.com/liferay/liferay-portal/blob/7.4.3.80-ga80/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/search/spi/model/index/contributor/JournalArticleModelDocumentContributor.java#L182-L186)
  - It eventually creates the keyword field by calling: [createKeywordField(name, value, false)](https://github.com/liferay/liferay-portal/blob/7.4.3.80-ga80/portal-kernel/src/com/liferay/portal/kernel/search/DocumentImpl.java#L391)
    - **Notice:** lowercase boolean is set to **false** [DocumentImpl::createField](https://github.com/liferay/liferay-portal/blob/7.4.3.80-ga80/portal-kernel/src/com/liferay/portal/kernel/search/DocumentImpl.java#L944-L952)
  - However, for the original sortable field, it does get normalized with lowercase: [_createSortableTextField(name, true, value)](https://github.com/liferay/liferay-portal/blob/7.4.3.80-ga80/portal-kernel/src/com/liferay/portal/kernel/search/DocumentImpl.java#L393)
    - We see it eventually calls: [createKeywordField(Field.getSortableFieldName(name), truncatedValue, true)](https://github.com/liferay/liferay-portal/blob/7.4.3.80-ga80/portal-kernel/src/com/liferay/portal/kernel/search/DocumentImpl.java#L1109-L1110)
    - **Notice:** lowercase boolean is set to **true** [DocumentImpl::createField](https://github.com/liferay/liferay-portal/blob/7.4.3.80-ga80/portal-kernel/src/com/liferay/portal/kernel/search/DocumentImpl.java#L944-L952)

**Headless API**
Headless API's filtering previously expected that the target field would be lowercase (since the *_sortable fields in Liferay are normalized to lowercase). With this in mind, when preparing their search query, the string that is provided is normalized to lowercase as well.
- Reference: https://issues.liferay.com/browse/LPS-85266
- Relevant code: [ExpressionVisitorImpl.java#L221-L226](https://github.com/liferay/liferay-portal/blob/7.4.3.80-ga80/modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImpl.java#L221-L226)